howso/update_cases.amlg

```diff
@@ -440,84 +440,112 @@
+											"fixed rand seed"
+											(null) ;radius
+											!numericalPrecision
+										)
+									))
+							))
+
+							;if there are perfect matches (value is infinite), keep only those and set their weight to 1
+							(if (contains_value (values closest_cases_map) .infinity)
+								;keep only perfect matches
+								(map 1 (filter (lambda (= (current_value) .infinity)) closest_cases_map) )
+
+								;else return map as-is
+								closest_cases_map
 							)
 						))
-				))
-
-				;if there are perfect matches (value is infinite), keep only those and set their weight to 1
-				(if (contains_value (values closest_cases_map) .infinity)
-					(assign (assoc
-						closest_cases_map
-							;keep only perfect matches
-							(map 1 (filter (lambda (= (current_value) .infinity)) closest_cases_map) )
-					))
-				)
+						batch_of_cases
+					)
+			))
 
-				(assign (assoc total_weight (apply "+" (values closest_cases_map))))
+			;distribute the weights by writing to each of the local models' cases
+			(map
+				(lambda (let
+					(assoc
+						;closest_cases_map (current_value 1)
+						case_weight
+							(if (!= (null) weight_feature_index)
+								(get batch_of_cases [ (current_index 1) weight_feature_index ] )
+								;assume weight of 1 if weight feature values are not given
+								1
+							)
+						total_weight (apply "+" (values (current_value 1)))
+					)
 
-				;iterate over all the closest cases and accumulate the normalized weight to each one
-				(map
-					(lambda (seq
-						;if accumulate_weight_feature label doesn't exist in this case, add it with a weight `1`
-						(if (not (contains_label (current_index) accumulate_weight_feature))
-							(accum_entity_roots (current_index) new_weight_label_and_value)
-						)
+					;iterate over all the closest cases and accumulate the normalized weight to each one
+					(map
+						(lambda
+							;if accumulate_weight_feature label doesn't exist in this case, add it with a weight `1`
+							(if (not (contains_label (current_index) accumulate_weight_feature))
+								(accum_entity_roots (current_index)
+									(zip_labels
+										(list accumulate_weight_feature)
+										(list (+ 1 (* (/ (current_value 1) total_weight) case_weight)) )
+									)
+								)
 
-						(accum_to_entities
-							(current_index)
-							(associate
-								accumulate_weight_feature
-									(* (/ (current_value 1) total_weight) case_weight)
+								;else label exists, accumulate to it
+								(accum_to_entities
+									(current_index)
+									(associate
+										accumulate_weight_feature
+											(* (/ (current_value 1) total_weight) case_weight)
+									)
+								)
 							)
 						)
-					))
-					closest_cases_map
-				)
-			))
-			cases
+						(current_value) ;closest_cases_map
+					)
+				))
+				closest_cases_maps
+			)
+
+			(accum (assoc case_index batch_size))
```

Comment:
on the accum below to !dataMassChanceSinceLastAnalyze, it uses `(size cases)`. I believe we need to sum the case weights if those values are specified in the case data. basically if `weight_feautre_index` is not `(null)`